pkg/build/strategies/layered/layered.go

```diff
@@ -63,14 +64,24 @@ func (b *Layered) CreateDockerfile(config *api.Config) error {
 	}
 
 	buffer.WriteString(fmt.Sprintf("FROM %s\n", b.config.BuilderImage))
-	buffer.WriteString(fmt.Sprintf("COPY scripts %s\n", locations[0]))
+	// if -s / --scripts / --scripts-url are not specified, there will not be a scripts subdir
+	// off of the working dir, so don't copy here, and we'll have to tweak the scripts URL to use the env var
+	// STI_SCRIPTS_PATH
+	scriptsIncluded := len(config.ScriptsURL) > 0
```

Comment:
OK ... the light bulb finally lit ... better understand what is going on.  The error message from the Dockerfile COPY is misleading.  It is not that the ./scripts dir is missing.  It is that it is empty !!!

So to tweak the suggestion from @mfojtik, I switched from examining ScriptsURL to checking for the existence of the required `assemble` script in the scripts directory.  Using that to avoid the COPY scripts when appropriate, and analogously setting ScriptsURL later on to either the existing image:// url on the STI_SCRIPTS_PATH env var have things working.

Now, I'm curious about whether:
a) does anything ever show up in the scripts dir with the layered path
b) should we (at least when the scripts dir is empty) just star calling InstallRequired / InstallOptional https://github.com/openshift/source-to-image/blob/master/pkg/scripts/install.go#L50-L69 to pull the scripts from wherever (including .sti/bin)